pkg/rpc/wrappers/block.go

```diff
@@ -1,34 +1,102 @@
+		Size       int                     `json:"size"`
+		Type       string                  `json:"type"`
+		Version    uint8                   `json:"version"`
+		Attributes []transaction.Attribute `json:"attributes"`
+		VIn        []transaction.Input     `json:"vin"`
+		VOut       []transaction.Output    `json:"vout"`
+		Scripts    []transaction.Witness   `json:"scripts"`
+
+		SysFee string `json:"sys_fee"`
+		NetFee string `json:"net_fee"`
+
+		Nonce uint32 `json:"nonce,omitempty"`
+	}
+
 	// Block wrapper used for the representation of
 	// block.Block / block.Base on the RPC Server.
 	Block struct {
-		*block.Block
-		Confirmations uint32       `json:"confirmations"`
-		NextBlockHash util.Uint256 `json:"nextblockhash,omitempty"`
-		Hash          util.Uint256 `json:"hash"`
+		Hash              util.Uint256 `json:"hash"`
+		Size              int          `json:"size"`
+		Version           uint32       `json:"version"`
+		NextBlockHash     util.Uint256 `json:"nextblockhash,omitempty"`
+		PreviousBlockHash util.Uint256 `json:"previousblockhash"`
+		MerkleRoot        util.Uint256 `json:"merkleroot"`
+		Time              uint32       `json:"time"`
+		Index             uint32       `json:"index"`
+		Nonce             string       `json:"nonce"`
+		NextConsensus     util.Uint160 `json:"nextconsensus"`
+
+		Confirmations uint32 `json:"confirmations"`
+
+		Script transaction.Witness `json:"script"`
+
+		Tx []Tx `json:"tx"`
 	}
 )
 
 // NewBlock creates a new Block wrapper.
-func NewBlock(block *block.Block, chain core.Blockchainer) Block {
-	blockWrapper := Block{
-		Block: block,
-		Hash:  block.Hash(),
+func NewBlock(b *block.Block, chain core.Blockchainer) Block {
+	res := Block{
+		Version:           b.Version,
+		Hash:              b.Hash(),
+		Size:              io.GetVarSize(b),
+		PreviousBlockHash: b.PrevHash,
+		MerkleRoot:        b.MerkleRoot,
+		Time:              b.Timestamp,
+		Index:             b.Index,
+		Nonce:             strconv.FormatUint(b.ConsensusData, 16),
+		NextConsensus:     b.NextConsensus,
+		Confirmations:     chain.BlockHeight() - b.Index - 1,
+
+		Script: b.Script,
+
+		Tx: make([]Tx, 0, len(b.Transactions)),
 	}
 
-	hash := chain.GetHeaderHash(int(block.Index) + 1)
+	hash := chain.GetHeaderHash(int(b.Index) + 1)
 	if !hash.Equals(util.Uint256{}) {
-		blockWrapper.NextBlockHash = hash
+		res.NextBlockHash = hash
+	}
+
+	for i := range b.Transactions {
+		tx := Tx{
+			TxID:       b.Transactions[i].Hash(),
+			Size:       io.GetVarSize(b.Transactions[i]),
+			Type:       b.Transactions[i].Type.String(),
+			Version:    b.Transactions[i].Version,
+			Attributes: b.Transactions[i].Attributes,
+			VIn:        b.Transactions[i].Inputs,
+			VOut:       b.Transactions[i].Outputs,
+			Scripts:    b.Transactions[i].Scripts,
+		}
+
+		tx.SysFee = chain.SystemFee(b.Transactions[i]).String()
+		tx.NetFee = chain.NetworkFee(b.Transactions[i]).String()
+
+		// set nonce only for MinerTransaction
```

Comment:
Do we need `omitempty` then?

Reply:
yep, for other tx's field should be omitted